xrayvision/visibility.py

```diff
@@ -336,11 +338,49 @@ def phase_uncertainty(self):
                     * apu.rad
                 ).to(apu.deg)
 
+    def index_by_label(self, labels: Iterable[str]):
+        """
+        Extract visibilities based on their labels.
+
+        Parameters
+        ----------
+        labels :
+            The labels of the desired visibilities.
+
+        Returns
+        -------
+        new_vis : Same as self type
+        """
+        self_labels = self.meta.vis_labels
+        if self_labels is None:
+            raise ValueError("self.meta.vis_labels must be set to index by label.")
+        idx = [np.where(self_labels == label)[0][0] for label in labels]
+        new_data = self._data.isel({self._uv_key: idx})
+        new_data.attrs[self._meta_key][_VIS_LABELS_KEY] = labels
+        new_vis = copy.deepcopy(self)
+        new_vis._data = new_data
+        return new_vis
+
     def _build_quantity(self, label, unit_label=None):
         if unit_label is None:
             unit_label = label
         return apu.Quantity(self._data[label], unit=self._data.attrs[self._units_key][unit_label])
 
+    def __getitem__(self, item):
+        if isinstance(item, numbers.Integral):
+            item = (slice(item, item + 1),)
+        if not isinstance(item, tuple):
+            item = (item,)
+        dims = self._data.dims
+        if len(item) != len(dims):
+            item = list(item) + [slice(None)] * (len(dims) - len(item))
```

Comment:
Does this assume the the uv axis is first?

As above we have
> `# In case visibilities is multi-dimensional, assume last axis is the uv-axis. `

Reply:
No.  If the user doesn't supply a slice for every dimension, the trailing dimensions are not sliced.  To make things easy, they are actually sliced by `slice(None)`, which comes to the same thing.  There is not assumption here about which axis corresponds to the uv axis.  However, there are a couple places elsewhere where I've assumed the uv axis is the last axis.  However, this could be changed.

Reply:
I guess all I meant is that is id did `vis[:2]` I'd expect to be slicing the vis/uv axis which in the case of 1d is true but not with higher dimensions.